toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1122,6 +1150,54 @@ func (g *PkgGraph) AddGoalNodeWithExtraLayers(goalName string, packages, tests [
 	return
 }
 
+// AddGoalNodeToNodes adds a goal node to the graph which links to a list of existing nodes.
+func (g *PkgGraph) AddGoalNodeToNodes(goalName string, existingNodes []*PkgNode, extraLayers int) (goalNode *PkgNode, err error) {
```

Comment:
There's already a function called `AddGoalNodeWithExtraLayers` above, with a quite similar list of parameters. Could you explain the difference between the two?